WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/Prompts/DashboardPromptsCardCell.swift

```diff
@@ -289,6 +289,26 @@ class DashboardPromptsCardCell: UICollectionViewCell, Reusable {
 
     // Defines the structure of the contextual menu items.
     private var contextMenuItems: [[MenuItem]] {
+        let viewMoreMenuTapped = { [weak self] in
+            guard let self else { return }
+            self.viewMoreMenuTapped()
+        }
+
+        let skipMenuTapped = { [weak self] in
+            guard let self else { return }
+            self.skipMenuTapped()
+        }
+
+        let learnMoreTapped = { [weak self] in
+            guard let self else { return }
+            self.learnMoreTapped()
+        }
+
+        let removeMenuTapped = { [weak self] in
+            guard let self else { return }
+            self.removeMenuTapped()
+        }
+
```

Comment:
> So the issue is that the cell and the button have a retain cycle, right?

Yes, more or less correct. `DashboardPromptCardCell` holds `BlogDashboardCardFrameView` which holds menu actions, that capture `DashboardPromptCardCell` strongly in a closure, creating a reference cycle.

Here's the memory graph after logging out if this change hadn't been made. You can see that `DashboardPromptCardCell` holds the reference to `BlogDashboardCardFrameView` and we we go through all the reference tree we can again reach `DashboardPromptCardCell` which is held strongly by a Swift closure.

https://github.com/wordpress-mobile/WordPress-iOS/assets/4062343/ac44cf1c-915e-40de-a6ce-335e5f47b4dc

Reply:
The lesson here is to avoid assigning functions to closure as a parameter. It is easy to mismanage memory this way https://www.marisibrothers.com/2017/04/memory-leak-in-swift-assigning-function.html

Reply:
Until now I've thought that assigning a function to a callback closure was neat, but TIL it creates a retain cycle 😬 Thanks for catching this.

Reply:
It is neat 😁
It's usually OK if you are assigning a ViewModel's method:

```swift
self.button.action = self.viewModel.buttonTapped
```

The views are expected to retain their ViewModels.

Reply:
Yes, that's correct. However, when we don't have a clear architecture in most of the places in the code, we need to be careful with this.